pandora-server-shard/src/networking/manager_client.ts

```diff
@@ -414,6 +416,184 @@ export const ConnectionManagerClient = new class ConnectionManagerClient impleme
+												space.handleActionMessage({
+													id: 'gamblingCardGameDealPlayerOpen',
+													character,
+													sendTo: space.cardGame.getPlayerIds(), // Send messages only to players
+													dictionary: {
+														'TARGET_CHARACTER': `${space.getCharacterById(game.targetId)?.name}`,
+														'CARD': card.toString(),
+													},
+												});
+											} else {
+												space.handleActionMessage({
+													id: 'gamblingCardGameDealPlayerSecret',
+													character,
+													sendTo: space.cardGame.getPlayerIds(), // Send messages only to players
+													dictionary: {
+														'TARGET_CHARACTER': `${space.getCharacterById(game.targetId)?.name}`,
+													},
+												});
+												space.handleActionMessage({
+													id: 'gamblingCardGameDealToYou',
+													character,
+													sendTo: [game.targetId],
+													dictionary: {
+														'CARD': card.toString(),
+													},
+												});
+											}
+										} else {
+											//Dealt to the room openly
+											space.handleActionMessage({
+												id: 'gamblingCardGameDealOpen',
+												character,
+												sendTo: space.cardGame.getPlayerIds(), // Send messages only to players
+												dictionary: {
+													'CARD': card.toString(),
+												},
+											});
+										}
+									}
+								} else {
+									space.handleActionMessage({
+										id: 'gamblingCardGameNotAllowed',
+										sendTo: [client.character.id],
+										character,
+									});
+								}
+								break;
+							}
+							case 'check': {
+								space.handleActionMessage({
+									id: 'gamblingCardGameHandCheck',
+									character,
+									sendTo: [client.character.id],
+									dictionary: {
+										'HAND': space.cardGame.getPlayerHand(client.character.id),
+									},
+								});
+								break;
+							}
+							case 'reveal': {
+								if (space.cardGame.isDealer(client.character.id)) {
+									// Send the cards of all players to the players and end the game
+									const players = space.cardGame.getPlayerIds();
+									const spaceHand = space.cardGame.getSpaceHand();
+									// Remind players of the cards on the table
+									space.handleActionMessage({
+										id: 'gamblingCardGameRoomCards',
+										character,
+										sendTo: players,
+										dictionary: {
+											'HAND': space.cardGame.getSpaceHand(),
+											'ISARE': (spaceHand === 'nothing' || spaceHand.length < 3) ? 'is' : 'are',
+										},
+									});
+									//Show the crads of all players
+									players.forEach((id) => space.handleActionMessage({
+										id: 'gamblingCardGameHandReveal',
+										character,
+										sendTo: players,
+										dictionary: {
+											'PLAYER': `${space.getCharacterById(id)?.name}`,
+											'HAND': `${space.cardGame?.getPlayerHand(id)}`,
+										},
+									}));
+									space.cardGame = null;
```

Comment:
[note] It honestly feels a bit weird that doing reveal stops the game, mainly because it forces all the players to re-register. But I think that that could be improved in future versions.